src/wallet/wallet.cpp

```diff
@@ -1625,6 +1666,8 @@ CBlockIndex* CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, bool f
         ShowProgress(_("Rescanning..."), 100); // hide progress dialog in GUI
 
         fScanningWallet = false;
+
+        m_last_block_processed = chainActive.Tip();
```

Comment:
This will be set incorrectly if the rescan was aborted.

Reply:
I just removed this and set it to chainActive.Tip() in CreateWalletFromFile. It doesnt make sense to try to use this to block wallet until we're caught up with the current tip in reorg, it isnt a regression so we should just leave it. When/if we expose last_block_processed via RPC (and allow wallet to return data that is stale) we should change that, I suppose.

Reply:
Yes - I agree. `m_last_block_processed` would be a really nice value to expose in the `getwalletinfo` RPC.

I have a weak preference to make `m_last_block_processed` work as you'd expect (tracking the last block processed by the wallet), but this PR is already doing a lot, so I think it's also fine to keep this as is. Please update the comment in wallet.h to say that it's not actually tracking how far the wallet is behind chain sync and shouldn't be publicly exposed until that's changed.

Reply:
OK, added more info to the wallet.h comment.